src/flb_log_event_decoder.c

```diff
@@ -261,21 +262,36 @@ int flb_event_decoder_decode_object(struct flb_log_event_decoder *context,
     return FLB_EVENT_DECODER_SUCCESS;
 }
 
+int flb_log_event_decoder_get_decode_result(struct flb_log_event_decoder *context)
+{
+    if (context->decode_result == FLB_EVENT_DECODER_ERROR_INSUFFICIENT_DATA &&
+        context->offset == context->length) {
+        context->decode_result = FLB_EVENT_DECODER_SUCCESS;
+    }
+
+    return context->decode_result;
+}
+
 int flb_log_event_decoder_next(struct flb_log_event_decoder *context,
                                struct flb_log_event *event)
 {
     size_t previous_offset;
     int    result;
 
-    context->record_base = NULL;
-    context->record_length = 0;
-
     if (context == NULL) {
         return FLB_EVENT_DECODER_ERROR_INVALID_CONTEXT;
     }
+    if (context->length == 0) {
+        context->decode_result = FLB_EVENT_DECODER_ERROR_NO_INCOMING_DATA;
+        goto flb_log_event_decoder_next_end;
+    }
```

Comment:
If `context->length` is zero, `flb_log_event_decoder_get_decode_result` will return `FLB_EVENT_DECODER_SUCCESS`.
This if block is to prevent it.

Reply:
Remove `FLB_EVENT_DECODER_ERROR_NO_INCOMING_DATA` and use `FLB_EVENT_DECODER_ERROR_INSUFFICIENT_DATA` instead.

Reply:
I fixed it.